main.tf

```diff
@@ -81,14 +81,23 @@ resource "aws_wafv2_web_acl" "main" {
 # WAFv2 web acl association with ALB
 #####
 resource "aws_wafv2_web_acl_association" "main" {
-  count = var.enabled && var.create_alb_association ? 1 : 0
+  count = var.enabled && var.create_alb_association && var.alb_arn != "" ? 1 : 0
 
   resource_arn = var.alb_arn
   web_acl_arn  = aws_wafv2_web_acl.main[0].arn
 
   depends_on = [aws_wafv2_web_acl.main]
 }
 
+resource "aws_wafv2_web_acl_association" "alb_list" {
+  count = var.enabled && var.create_alb_association ? length(var.alb_arn_list) : 0
```

Comment:
perhaps in this `count` we won't need `var.create_alb_association` and/or it should be replaced by `length(var.alb_arn_list) > 0` something to test

Reply:
I fixed the initial issue and decided to keep `create_alb_association`, hope you don't mind